cmd/soroban-cli/src/commands/contract/info/shared.rs

```diff
@@ -0,0 +1,82 @@
+use std::path::PathBuf;
+
+use clap::arg;
+use soroban_env_host::xdr;
+use soroban_rpc::Client;
+
+use crate::commands::contract::info::shared::Error::InvalidWasmHash;
+use crate::commands::contract::InfoOutput;
+use crate::config::{locator, network};
+use crate::utils::rpc::get_remote_wasm_from_hash;
+use crate::wasm;
+use crate::wasm::Error::ContractIsStellarAsset;
+
+#[derive(Debug, clap::Args, Clone, Default)]
+#[command(group(
+    clap::ArgGroup::new("src")
```

Comment:
I suggest naming the group a full word captialized because I'm pretty sure they get rendered alongside other group names that are by default cased, e.g. `Options`.
```suggestion
    clap::ArgGroup::new("Source")
```

Unclear to me if the source grouping is of significant value. The source grouping is not something the cli uses on other similar commands. But we could update those other places to use it. If we're going to do it we should make sure it works well consistently in all similar commands, so I suggest leaving off the group in this PR, and following up after this PR with a holistic update of all similar commands so we can validate that a holistic story for it is good.

Reply:
Oh I was using the group to make only one selection option to be valid. I guess we can just use conflicts-with and required instead? Personally, groups are more readable to me than mix of `conflicts-with` + `required_unless_present`, but I can stick to just that parameters combination instead ^

Reply:
Ah, I didn't realise this was another way to do that. I think that's fine, would be good to maintain some consistency so maybe an issue to follow up changing other commands to do same.

Reply:
Unfortunately, not very flexible when using derive API 😔 

> Network configuration is mandatory, even when fetching from local. This is due to group can't be marked as optional with clap's derive API. We can fix that in the future, potentially using more flexible builder API instead.